content-src/lib/selectLayoutRender.js

```diff
@@ -1,79 +1,70 @@
-import {createSelector} from "reselect";
+export const selectLayoutRender = (state, rickRollCache) => {
+  const {layout, feeds, spocs} = state;
+  let spocIndex = 0;
 
-export const selectLayoutRender = createSelector(
-  // Selects layout, feeds, spocs so that we only recompute if
-  // any of these values change.
-  [
-    state => state.DiscoveryStream.layout,
-    state => state.DiscoveryStream.feeds,
-    state => state.DiscoveryStream.spocs,
-  ],
-
-  // Adds data to each component from feeds. This function only re-runs if one of the inputs change.
-  // TODO: calculate spocs
-  function layoutRender(layout, feeds, spocs) {
-    let spocIndex = 0;
-
-    function maybeInjectSpocs(data, spocsConfig) {
-      if (data &&
-          spocsConfig && spocsConfig.positions && spocsConfig.positions.length &&
-          spocs.data.spocs && spocs.data.spocs.length) {
-        const recommendations = [...data.recommendations];
-        for (let position of spocsConfig.positions) {
-          let rickRoll = Math.random();
-          if (spocs.data.spocs[spocIndex] && rickRoll <= spocsConfig.probability) {
-            recommendations.splice(position.index, 0, spocs.data.spocs[spocIndex++]);
-          }
+  function maybeInjectSpocs(data, spocsConfig) {
+    if (data &&
+        spocsConfig && spocsConfig.positions && spocsConfig.positions.length &&
+        spocs.data.spocs && spocs.data.spocs.length) {
+      const recommendations = [...data.recommendations];
+      for (let position of spocsConfig.positions) {
+        // Cache random number for a position
+        if (!rickRollCache[position.index]) {
+          rickRollCache[position.index] = Math.random();
```

Comment:
Cache random value for a spoc position, so that it can be used on next render caused by a user interaction such as dismiss. We are resetting this cache on page refresh.